src/libImaging/BoxBlur.c

```diff
@@ -266,16 +266,16 @@ ImagingBoxBlur(Imaging imOut, Imaging imIn, float radius, int n) {
     /* Apply blur in one dimension.
        Use imOut as a destination at first pass,
        then use imOut as a source too. */
-    ImagingHorizontalBoxBlur(imOut, imIn, radius);
+    ImagingHorizontalBoxBlur(imOut, imIn, xradius);
     for (i = 1; i < n; i++) {
-        ImagingHorizontalBoxBlur(imOut, imOut, radius);
+        ImagingHorizontalBoxBlur(imOut, imOut, xradius);
```

Comment:
We could skip passes if it’s radius == 0

Reply:
Sure. I've pushed a commit